src/HttpMethodsClient.php

```diff
@@ -22,19 +25,29 @@ final class HttpMethodsClient implements HttpMethodsClientInterface
      */
     private $requestFactory;
 
+    /**
+     * @var StreamFactoryInterface|null
+     */
+    private $streamFactory;
+
     /**
      * @param RequestFactory|RequestFactoryInterface
      */
-    public function __construct(ClientInterface $httpClient, $requestFactory)
+    public function __construct(ClientInterface $httpClient, $requestFactory, StreamFactoryInterface $streamFactory = null)
     {
         if (!$requestFactory instanceof RequestFactory && !$requestFactory instanceof RequestFactoryInterface) {
             throw new \TypeError(
                 sprintf('%s::__construct(): Argument #2 ($requestFactory) must be of type %s|%s, %s given', self::class, RequestFactory::class, RequestFactoryInterface::class, get_debug_type($requestFactory))
             );
         }
 
+        if (!$requestFactory instanceof RequestFactory && null === $streamFactory) {
+            @trigger_error(sprintf('Passing a %s without a %s to %s::__construct() is deprecated as of version 2.3 and will be disallowed in version 3.0. A stream factory is required to create a request with a string body.', RequestFactoryInterface::class, StreamFactoryInterface::class, self::class));
```

Comment:
i assume that in next major, we might want to only support the psr factory, to move forward? then we could simply make the stream factory a required argument... (no need to change anything here, just a thought ;-) )

Reply:
I think so too. I am thinking once the old interfaces are all marked as deprecated in the future (https://github.com/php-http/message-factory/blob/master/src/ResponseFactory.php doesn't have a deprecation annotation yet), we can come back here to this constructor and also add a futher deprecation warning for if a psr request factory is not passed. Then yeh, in the next major version make all prams typed with psr interfaces, and all required.